### PR TITLE
fix: repair the app_protect_metadata

### PR DESCRIPTION
**Pull request by edarzins**
The `precompiledPublication` value was not being updated in the app_protect_metadata.json file.

### Proposed changes

The recent refactoring of plugins to extensions caused the updating of the app_protect_metadata.json file to be incomplete. This fixes the problem.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))

@dhurley

### CODE REVIEW

File: src/plugins/nginx.go [hunk 1]
```diff
@@ -639,3 +641,15 @@ func isFileInDirectoryMap(directoryMap *proto.DirectoryMap, path string) bool {
 	}
 	return false
 }
+
+func isNginxAppProtectEnabled(loadedConfig *config.Config) bool {
```
Comment (dhurley) [inline src/plugins/nginx.go:645]: Could you move this function to `src/core/config/types.go` and make it more generic by calling it something like `IsExtensionEnabled`.
Something similar to the `IsFeatureEnabled` function that we have https://github.com/nginx/agent/blob/main/src/core/config/types.go#L40

**Review by dhurley — COMMENTED** [review]
Could you also run `make install-tools` & `make deps`?

**Comment by dhurley** [comment]
Also looks like you need to make some updates to the `nginx_test.go` unit tests.

**Comment by oliveromahony** [comment]
can you run ```make deps``` as well. If you run ```make install-tools``` it should be in your git hooks then

File: src/plugins/nginx.go (hunk 1 above)
Comment (edarzins) [reply src/plugins/nginx.go]: Done

**Review by Dean-Coakley — APPROVED** [review]
Tested as working on `ubuntu:22.04`.

Deps up to date.

LGTM. Thanks @edarzins !

**Review by dhurley — APPROVED** [review]

**Review by oliveromahony — APPROVED** [review]

**Pull request merged**